data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -22,17 +47,155 @@
+        properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getGroupId());
+        schemaType = getSchemaType(sourceConfig.getSchemaConfig().getRegistryURL(), topicConfig.getName(), sourceConfig.getSchemaConfig().getVersion());
+        if (schemaType.isEmpty()) {
+            schemaType = MessageFormat.PLAINTEXT.toString();
+        }
+        setPropertiesForSchemaType(properties, schemaType);
+        if (sourceConfig.getAuthType().equalsIgnoreCase(AuthenticationType.PLAINTEXT.toString())) {
+            setPropertiesForAuth(properties);
+        }
+        return properties;
+    }
 
-  private static final Logger LOG = LoggerFactory.getLogger(KafkaSource.class);
-  
-  @Override
-  public void start(Buffer<Record<Object>> buffer) {
-    LOG.info("Initiated the start() method...");
-  }
+    private void setPropertiesForSchemaType(Properties properties,final String schemaType) {
+        if (schemaType.equalsIgnoreCase(MessageFormat.JSON.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+        } else if (schemaType.equalsIgnoreCase(MessageFormat.PLAINTEXT.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+        } else if (schemaType.equalsIgnoreCase(MessageFormat.AVRO.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                    KafkaAvroDeserializer.class);
+            properties.put(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, sourceConfig.getSchemaConfig().getRegistryURL());
+        }
+    }
 
-  @Override
-  public void stop() {
-   LOG.info("Initiated the stop method...");
-  }
+    private void setPropertiesForAuth(Properties properties) {
+        String username = sourceConfig.getAuthConfig().getPlainTextAuthConfig().getUsername();
+        String password = sourceConfig.getAuthConfig().getPlainTextAuthConfig().getPassword();
+        properties.put("sasl.mechanism", "PLAIN");
+        properties.put("sasl.jaas.config", "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + username + "\" password=\"" + password + "\";");
+        properties.put("security.protocol", "SASL_PLAINTEXT");
+    }
 
+    private static String getSchemaType(final String registryUrl, final String topicName, final int schemaVersion) {
+        StringBuilder response = new StringBuilder();
+        String schemaType = "";
+        try {
+            String urlPath = registryUrl + "subjects" + "/" + topicName + "-value/versions/" + schemaVersion;
```

Comment:
registryUrl should be validated prior to connection. is it expected to end with "/"? why is schemaVersion required? Can we not get the latest version? or list versions first and the read latest version.